gnocchi/rest/api.py

```diff
@@ -1890,6 +1899,102 @@ class BatchController(object):
     resources = ResourcesBatchController()
 
 
+class PrometheusWriteController(rest.RestController):
+
+    PROMETHEUS_RESOURCE_TYPE = {
+        "instance": {"type": "string",
+                     "min_length": 1,
+                     "max_length": 512,
+                     "required": True},
+        "job": {"type": "string",
+                "min_length": 1,
+                "max_length": 512,
+                "required": True}
+    }
+
+    @classmethod
+    @tenacity.retry(retry=tenacity.retry_if_exception_type(
```

Comment:
You don't want to retry forever. We need a time limit.

I think we have a time limit for refresh already. We can use that.

Reply:
If that retry more that once, I prefer not to be guy that do the sql maintenance operation in this case :), Fixed